File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;
class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        HashMap<String, Integer> map = new HashMap<>();
        HashMap<String, ArrayList<String>> reportMap = new HashMap<>();
        for(int i = 0; i < id_list.length; i++) {
            map.put(id_list[i], i); // index용
            reportMap.put(id_list[i], new ArrayList<>()); //신고 결과 저장
        }
        
        for(int i = 0; i < report.length; i++) {
            String[] str = report[i].split(" ");
            if(!reportMap.get(str[1]).contains(str[0])) {
                reportMap.get(str[1]).add(str[0]);
            }
        }
        
        for(String str : reportMap.keySet()) {
            if (k <= reportMap.get(str).size()) {
                for (String rep : reportMap.get(str)) {
                    answer[map.get(rep)]++;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    var answer = [];
    const reportList = {};
    const reportedNum = {};
    for (const id of id_list) {
        reportList[id] = new Set();
        reportedNum[id] = 0;
    }
    for (const r of report) {
        const [u, v] = r.split(' ');
        reportList[u].add(v);
    }
    for (const u in reportList) {
        for (const v of reportList[u]) {
            reportedNum[v]++;
        }
    }
    let idx = 0;
    for (const u in reportList) {
        let count = 0;
        for (const v of reportList[u]) {
            if (reportedNum[v] >= k) count++;
        }
        answer[idx++] = count;
        
    }
    return answer;
}